File: 프로그래머스/lv0/120852. 소인수분해/소인수분해.py
def solution(n):
    answer = set()
    s = [2, 3]

    for i in range(5, 10000):
        for j in s:
            if i % j == 0:
                break;
        else:
            s.append(i)
    
    idx = 0
    while n // s[idx] != 0:
        if n % s[idx] == 0:
            answer.add(s[idx])
            n //= s[idx]
        elif n % s[idx] != 0:
            idx += 1
    return sorted(list(answer))

File: 프로그래머스/lv0/120852. 소인수분해/소인수분해.java
import java.util.HashSet;
import java.util.ArrayList;
import java.util.Collections;

class Solution {
    public ArrayList<Integer> solution(int n) {
        HashSet<Integer> numList = new HashSet<Integer>();
        int num = 2;
        while(n != 1) {
            if(n % num == 0) {
                numList.add(num);
                n /= num;
            }
            else {
                num++;
            }
        }
        ArrayList<Integer> answer = new ArrayList<>(numList);
        Collections.sort(answer);
        return answer;
    }
}